lang/csharp/src/apache/main/Generic/GenericEnum.cs

```diff
@@ -37,8 +37,15 @@ public class GenericEnum
             get { return value; }
             set
             {
-                if (! Schema.Contains(value)) throw new AvroException("Unknown value for enum: " + value + "(" + Schema + ")");
-                this.value = value;
+                if (!string.IsNullOrEmpty(Schema.Default))
+                {
+                    this.value = Schema.Contains(value) ? value : Schema.Default;
+                }
+                else
+                {
+                    if (! Schema.Contains(value)) throw new AvroException("Unknown value for enum: " + value + "(" + Schema + ")");
```

Comment:
nit: might as well get rid of the space after the "!" to match the convention in the rest of the code.